src/ec2/instanceStateManager.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Ec2Client } from '../shared/clients/ec2Client'
+import { ToolkitError, isAwsError } from '../shared/errors'
+import { showMessageWithCancel } from '../shared/utilities/messages'
+import { Timeout } from '../shared/utilities/timeoutUtils'
+import { Ec2Selection } from './utils'
+
+export class InstanceStateManager {
```

Comment:
ins "Manager" communicating anything useful in these class names?

```suggestion
export class InstanceState {
```

Reply:
Instance state made me think of the physical state itself, whereas manager makes me think of "managing" that state. Do you think its clearer to leave out the Manager piece? Is another suffix that you see as more appropriate?